tutorials/src/bin/basic-tutorial-9.rs

```diff
@@ -0,0 +1,150 @@
+extern crate failure;
+use failure::Error;
+
+extern crate glib;
+extern crate gstreamer as gst;
+extern crate gstreamer_pbutils as pbutils;
+use pbutils::prelude::*;
+use pbutils::Discoverer;
+use pbutils::DiscovererContainerInfo;
+use pbutils::DiscovererInfo;
+use pbutils::DiscovererResult;
+use pbutils::DiscovererStreamInfo;
+
+use std::env;
+
+fn print_stream_info(info: &DiscovererStreamInfo, depth: usize) {
+    print!("{:indent_level$}", " ", indent_level = 2 * depth);
+    print!("{}:", info.get_stream_type_nick());
+
+    if let Some(caps) = info.get_caps() {
+        let desc = if caps.is_fixed() {
+            pbutils::pb_utils_get_codec_description(&caps)
+                .unwrap_or_else(|| String::from("unknown codec"))
+        } else {
+            caps.to_string()
+        };
+        println!(" {}", desc);
+    } else {
+        println!();
+    }
```

Comment:
This would probably be nicer with building a single string and then printing it all at once. E.g.

```rust
let caps_str = if let Some(caps) ... {
    ...
} else {
    String::new()
};

println!(....);
```